src/app/core/common-components/json-editor/json-editor.component.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  Output,
+  ViewChild,
+} from "@angular/core";
+import { CommonModule } from "@angular/common";
+import { MatButtonModule } from "@angular/material/button";
+import { createJSONEditor } from "vanilla-jsoneditor/standalone.js";
+import { AlertService } from "../../alerts/alert.service";
+
+/**
+ * Component for editing JSON data.
+ */
+@Component({
+  selector: "app-json-editor",
+  standalone: true,
+  imports: [CommonModule, MatButtonModule],
+  templateUrl: "./json-editor.component.html",
+  styleUrl: "./json-editor.component.scss",
+})
+export class JsonEditorComponent {
+  /**
+   * JSON value to be edited by the user.
+   */
+  @Input() value: object = {};
+
+  @Input() height = "65vh";
+
+  @Output() valueChange = new EventEmitter<any>();
+
+  @ViewChild("json", { static: true }) json!: ElementRef<HTMLDivElement>;
+
+  private jsonEditor: any;
+
+  constructor(private alertService: AlertService) {}
+
+  /**
+   * Initialize the JSON editor.
+   * This method is called after the component view is initialized.
+   * creates a JSON editor instance and sets up the onChange event handler.
+   */
+  ngAfterViewInit(): void {
+    this.jsonEditor = createJSONEditor({
+      target: this.json.nativeElement,
+      props: {
+        content: { json: this.value || {} },
+        onChange: (updatedContent: any, { contentErrors }: any) => {
+          if (contentErrors?.length === 0) {
+            this.value = updatedContent.json;
+          }
```

Comment:
What happen when contentErrors > 0? Should we display a notice here to the user?

Reply:
If `contentErrors > 0,` then something is wrong in JSON. I have made a few changes in the code to handle this error. Can you please take a look and let me know if any changes are needed?

Reply:
Thanks for this suggestion, it makes sense. I have updated it based on your suggestions. Now, the save button can only be pressed if the JSON is valid. Otherwise, the save button will remain disabled if the JSON is invalid.

**For reference:** I made some changes to the JSON to make it invalid, and the JSON editor component displayed an error indicating that the JSON is invalid and disabled the save button accordingly.

<img width="1254" alt="image" src="https://github.com/user-attachments/assets/401e5d36-6008-4ff0-b28e-41c63dfde252" />

Reply:
:+1: 

You can remove the alert. The red error bar in the editor is very visible and should be sufficient.